dashboard/app/models/user_level.rb

```diff
@@ -171,14 +172,18 @@ def self.update_lockable_state(user_id, level_id, script_id, locked, readonly_an
     # no need to create a level if it's just going to be locked
     return if !user_level.persisted? && locked
 
-    ## TODO use the locked setter here
-
-    user_level.assign_attributes(
-      readonly_answers: !locked && readonly_answers,
-      unlocked_at: locked ? nil : Time.now,
-      # level_group, which is the only levels that we lock, always sets best_result to 100 when complete
-      best_result: (locked || readonly_answers) ? ActivityConstants::BEST_PASS_RESULT : user_level.best_result
-    )
+    # we explicitly set `locked=false` if `readonly_answers=true` to start
+    # the autolock clock so that the mutually-exclusive `show_as_locked` and
+    # `show_as_readonly` will flip from `false/true` to `true/false`
+    # after AUTOLOCK_PERIOD has passed.
+    user_level.locked = readonly_answers ? false : locked
+    user_level.readonly_answers = readonly_answers
+
+    # level_group, which is the only type of level that we lock, always sets
+    # best_result to 100 when complete
+    if locked || readonly_answers
```

Comment:
minor cleanup here -- no need to set `best_result` at all if we're just going to set it to itself

Reply:
🙌